src/analysis/retail/priest/discipline/modules/guide/VoidSummonerBetweenCasts.tsx

```diff
@@ -0,0 +1,136 @@
+import spells from 'common/SPELLS/priest';
+import talents from 'common/TALENTS/priest';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events, { CastEvent } from 'parser/core/Events';
+import { explanationAndDataSubsection } from 'interface/guide/components/ExplanationRow';
+import { GUIDE_CORE_EXPLANATION_PERCENT } from '../../Guide';
+import SpellUsable from '../core/SpellUsable';
+import SpellLink from 'interface/SpellLink';
+import { QualitativePerformance } from 'parser/ui/QualitativePerformance';
+import { PerformanceBoxRow } from 'interface/guide/components/PerformanceBoxRow';
+
+// TODO: Ask Priest guide maker about these numbers
+const CAST_THRESHOLD_OK = 5;
+const CAST_THRESHOLD_GOOD = 9;
+
+class VoidSummonerBetweenCasts extends Analyzer {
+  static dependencies = {
+    spellUsable: SpellUsable,
+  };
+  protected spellUsable!: SpellUsable;
+
+  _lastCast: number = 0;
+  _castsPerCD: {
+    timestamp: number;
+    casts: number;
+  }[] = [];
+  _totalSquidCasts: number = 0;
+
+  _currentSquid:
+    | typeof talents.MINDBENDER_DISCIPLINE_TALENT
+    | typeof talents.SHADOWFIEND_TALENT
+    | typeof talents.VOIDWRAITH_TALENT;
+
+  constructor(options: Options) {
+    super(options);
+
+    this.addEventListener(
```

Comment:
Check if the talent is active, if not return before assigning any events. We do do this like so;
```tsx
this.active = this.selectedCombatant.hasTalent(....Talent);
if (!this.active) {
  return;
}
```